core/src/main/scala/org/apache/spark/h2o/utils/H2OContextRestAPIUtils.scala

```diff
@@ -40,6 +43,29 @@ trait H2OContextRestAPIUtils extends H2OContextUtils {
     getCloudInfoFromNode(endpoint, conf)
   }
 
+  def downloadLogs(destinationDir: String, logContainer: String, conf: H2OConf): String = {
+    val endpoint = getClusterEndpoint(conf)
+    val file = new File(destinationDir, s"${logFileName()}.${logContainer.toLowerCase}")
+    logContainer match {
+      case "LOG" =>
+        val out = new java.io.FileWriter(file)
+        val content = readStringURLContent(endpoint, s"3/Logs/download/$logContainer", conf)
+        out.write(content)
```

Comment:
Good questions -> I can image case when H2O cluster would run several days in DEBUG mode that this would really cause OOM. I have tried to implement more memory-friendly solution. Please have a look